src/curl.c

```diff
@@ -118,6 +118,31 @@ int swupd_curl_init(void)
 	return 0;
 }
 
+/* 
+ * Cannot avoid a TOCTOU here with the current curl API.  Only using the curl
+ * API, a curl_easy_setopt does not detect if the client SSL certificate is
+ * present on the filesystem.  The only happens during curl_easy_perform.
+ * The emphasis is rather on how using a SSL client certificate is an opt-in
+ * function rather than an opt-out function.
+ */
+CURLcode swupd_curl_set_optional_client_cert(CURL* curl) {
+	CURLcode curl_ret = CURLE_OK;
+
+	if (access(SSL_CLIENT_CERT, F_OK) != -1) {
```

Comment:
Why not just check if `access(...) == 0`? It seems backwards to check if `access` returns not an error.

Reply:
Makes sense.  I'll go ahead and change this.